exercises/pascals-triangle/src/example.c

```diff
@@ -0,0 +1,26 @@
+#include "pascals_triangle.h"
+
+long long unsigned int factorial(int n)
+{
+   return (n == 0) ? 1 : n * factorial(n - 1);
+}
+
+long long unsigned int combination(int n, int r)
+{
+   return (r > n) ? 0 : factorial(n) / (factorial(n - r) * factorial(r));
+}
+
+int pascals_triangle(int count, int input[][((count < 0) ? 1 : count)])
+{
+   if (count < 0) {
+      memset(input, -1, sizeof(int) * count * count);
```

Comment:
if count is less than 0 then `sizeof(int) * count * count` will not be very good, though it will probably work in most cases. For example, if count is `-20` then this `memset()` will try and clear 400 int's worth of space... it only needs 1 ints worth.

 `sizeof(int)` would be better. Or `sizeof(input[0][0])` would be even better.